models/issues/comment.go

```diff
@@ -1161,14 +1161,9 @@ func DeleteComment(ctx context.Context, comment *Comment) error {
 // UpdateCommentsMigrationsByType updates comments' migrations information via given git service type and original id and poster id
 func UpdateCommentsMigrationsByType(ctx context.Context, tp structs.GitServiceType, originalAuthorID string, posterID int64) error {
 	_, err := db.GetEngine(ctx).Table("comment").
-		Where(builder.In("issue_id",
-			builder.Select("issue.id").
-				From("issue").
-				InnerJoin("repository", "issue.repo_id = repository.id").
-				Where(builder.Eq{
-					"repository.original_service_type": tp,
-				}),
-		)).
+		Join("INNER", "issue", "issue.id = comment.issue_id").
+		Join("INNER", "repository", "issue.repo_id = repository.id").
+		Where("repository.original_service_type = ?", tp).
 		And("comment.original_author_id = ?", originalAuthorID).
 		Update(map[string]any{
```

Comment:
I thought SQLite does not support `UPDATE JOIN`?
Or does xorm automatically generate the correct query?

Reply:
Or was it that only `xorm does not support UPDATE JOIN on SQLite`?

Reply:
https://gitea.com/xorm/xorm/pulls/2383 looks like it was just blocked by xorm

Reply:
Yes, but does this work under SQLite?
I've just looked through that PR, and to me, it looks like for SQLite, still an `UPDATE JOIN` is generated, even though that is not supported AFAIK.

Reply:
https://www.sqlite.org/lang_update.html i would say as long as you have the join-select within the where it works - through the changes of xorm dont reflect that yes

tests pass ?!? dont we have coverage of it?!?

Reply:
per https://www.sqlite.org/lang_update.html, sqlite support `UPDATE table1 SET xxx FROM table2 WHERE join conditions` from 3.33.0(2020-8-14).